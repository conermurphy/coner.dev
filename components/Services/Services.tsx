import React from 'react';
import { MdCode, MdKeyboard, MdShare } from 'react-icons/md';
import { Service } from '../../types';

interface IProps {
  services: Service[];
}

function getIcon(icon: string) {
  switch (icon) {
    case 'keyboard':
      return <MdKeyboard size="1.5rem" />;
    case 'code':
      return <MdCode size="1.5rem" />;
    case 'socials':
      return <MdShare size="1.5rem" />;
    default:
      break;
  }
}

export default function Services({ services }: IProps): JSX.Element {
  return (
    <div className="flex flex-col items-center w-100-perc my-10 md:my-72">
      <div className="max-w-[1334px] md:px-20 lg:px-106">
        <h2 className="font-bold text-32 md:text-40">My Crafts...</h2>
        <p className="text-lg md:text-2xl opacity-75 mb-10">
          Let me help you, hereâ€™s how.
        </p>
        <div className="flex flex-col items-center flex-wrap gap-12 xl:gap-[78px] md:flex-row lg:flex-nowrap">
          {services.map(({ title, copy, icon }) => {
            const iconSVG = getIcon(icon);
            return (
              <div key={title} className="max-w-xs">
                <div className="flex flex-row items-center gap-3 mb-5">
                  <div className="w-11 h-11 flex items-center justify-center bg-primary text-offWhite rounded-[10px]">
                    {iconSVG}
                  </div>
                  <h3 className="text-2xl font-semibold">{title}</h3>
                </div>
                <p className="opacity-75 max-w-[274px] lg:max-w-[320px]">
                  {copy}
                </p>
              </div>
            );
          })}
        </div>
      </div>
    </div>
  );
}
