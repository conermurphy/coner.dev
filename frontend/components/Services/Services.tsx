import React, { useEffect, useState } from 'react';
import { motion } from 'framer-motion';
import { Service } from '../../types';
import ComponentWrapper from '../ComponentWrapper/ComponentWrapper';
import FullStackDeveloperGraphic from '../Graphics/Fullstack-Developer';
import TechWriterGraphic from '../Graphics/Tech-Writer';
import VideoMakerGraphic from '../Graphics/Video-Maker';

interface IProps {
  services: Service[];
}

export default function Services({ services }: IProps): JSX.Element {
  const [activeService, setActiveService] = useState<Service>(services[0]);
  const [activeGraphic, setActiveGraphic] = useState<JSX.Element | null>(null);

  useEffect(() => {
    switch (activeService.title) {
      case 'Fullstack Developer':
        setActiveGraphic(<FullStackDeveloperGraphic />);
        break;
      case 'Technical Writer':
        setActiveGraphic(<TechWriterGraphic />);
        break;
      case 'Content Creator':
        setActiveGraphic(<VideoMakerGraphic />);
        break;
      default:
        break;
    }
  }, [activeService, services]);

  useEffect(() => {
    setInterval(() => {
      const currentServiceIndex = services.findIndex(
        (service) => service.title === activeService.title
      );

      const nextServiceIndex =
        currentServiceIndex + 1 > services.length - 1
          ? 0
          : currentServiceIndex + 1;

      setActiveService(services[nextServiceIndex]);
    }, 5000);
  }, [services, setActiveService, activeService]);

  return (
    <ComponentWrapper
      data={{
        title: 'What I Do',
        tag: 'About Me',
        description:
          'Lorem ipsum dolor sit amet consectetur. A arcu amet viverra et ullamcorper eget ac.',
      }}
    >
      <div className="grid grid-cols-1 lg:grid-cols-2 w-full justify-items-center items-center ">
        <ul className="max-w-md lg:max-w-2xl">
          {services.map((service) => (
            <motion.li
              key={service.title}
              className={`border-l-8 ${
                activeService.title === service.title
                  ? 'border-brand'
                  : 'border-text/10'
              } p-8 flex flex-col gap-2`}
            >
              <h3 className="font-heading font-extrabold text-text/90 text-xl lg:text-2xl">
                {service.title}
              </h3>
              <p className="md:text-lg">{service.copy}</p>
            </motion.li>
          ))}
        </ul>
        <div className="flex flex-row items-center justify-center h-64 w-64 lg:h-96 lg:w-96">
          {activeGraphic}
        </div>
      </div>
    </ComponentWrapper>
  );
}
