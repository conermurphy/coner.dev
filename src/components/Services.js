import React from 'react';
import styled from 'styled-components';
import { motion } from 'framer-motion';
import Services from '../data/services.json';

const ServiceSectionContainer = styled.section`
  background-color: var(--secondaryBg);
  width: 100vw;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  padding: 5rem 0;

  & > h2,
  h3 {
    text-align: center;
  }

  .ServiceCardsContainer {
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(clamp(300px, 50vw, 400px), 1fr));
    align-items: flex-start;
    gap: clamp(1rem, 2vw, 2rem);
    padding-top: 2rem;

    width: clamp(300px, 80vw, var(--maxWidth));
  }
`;

const ServiceCardContainer = styled(motion.div)`
  background-color: var(--primaryBg);
  filter: drop-shadow(var(--shadow));
  border-radius: var(--borderRadius);
  padding: 3rem;
  margin: 1rem;

  .serviceCardTitle {
    margin: 0;
    color: var(--primaryText);
    font-weight: 500;
    font-size: 2.5rem;
  }
`;

function ServiceCard({ service: { title, description } }) {
  return (
    <ServiceCardContainer>
      <h4 className="serviceCardTitle">{title}</h4>
      <p>{description}</p>
    </ServiceCardContainer>
  );
}

export function ServiceSection() {
  return (
    <ServiceSectionContainer id="services">
      <h2 className="title">Services</h2>
      <h3 className="subtitle">It's more than just a website...</h3>
      <div className="ServiceCardsContainer">
        {Services.map((service) => (
          <ServiceCard service={service} key={service.title} />
        ))}
      </div>
    </ServiceSectionContainer>
  );
}
