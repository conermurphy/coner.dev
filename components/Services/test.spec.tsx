import { render } from '@testing-library/react';
import React from 'react';
import Services from './Services';

const services = [
  {
    title: 'Service Title 1',
    copy: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla arcu at facilisis dignissim arcu pretium tempus. Cras scelerisque urna amet sagittis donec arcu ipsum.',
    icon: 'keyboard',
  },
  {
    title: 'Service Title 2',
    copy: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla arcu at facilisis dignissim arcu pretium tempus. Cras scelerisque urna amet sagittis donec arcu ipsum.',
    icon: 'code',
  },
  {
    title: 'Service Title 3',
    copy: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla arcu at facilisis dignissim arcu pretium tempus. Cras scelerisque urna amet sagittis donec arcu ipsum.',
    icon: 'socials',
  },
];

describe('Services', () => {
  it('Matches the snapshot', () => {
    const { container } = render(<Services services={services} />);
    expect(container).toMatchSnapshot();
  });
});
