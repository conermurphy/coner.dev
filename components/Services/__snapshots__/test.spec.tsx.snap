// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Services Matches the snapshot 1`] = `
<div>
  <div
    class="flex flex-col items-center w-100-perc my-10 md:my-72"
  >
    <div
      class="max-w-[1334px] md:px-20 lg:px-106"
    >
      <h2
        class="font-bold text-32 md:text-40"
      >
        My Crafts...
      </h2>
      <p
        class="text-lg md:text-2xl opacity-75 mb-10"
      >
        Let me help you, hereâ€™s how.
      </p>
      <div
        class="flex flex-col items-center flex-wrap gap-12 xl:gap-[78px] md:flex-row lg:flex-nowrap"
      >
        <div
          class="max-w-xs"
        >
          <div
            class="flex flex-row items-center gap-3 mb-5"
          >
            <div
              class="w-11 h-11 flex items-center justify-center bg-primary text-offWhite rounded-[10px]"
            >
              <svg
                fill="currentColor"
                height="1.5rem"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1.5rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20 5H4c-1.1 0-1.99.9-1.99 2L2 17c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm-9 3h2v2h-2V8zm0 3h2v2h-2v-2zM8 8h2v2H8V8zm0 3h2v2H8v-2zm-1 2H5v-2h2v2zm0-3H5V8h2v2zm9 7H8v-2h8v2zm0-4h-2v-2h2v2zm0-3h-2V8h2v2zm3 3h-2v-2h2v2zm0-3h-2V8h2v2z"
                />
                <path
                  d="M0 0h24v24H0zm0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </div>
            <h3
              class="text-2xl font-semibold"
            >
              Service Title 1
            </h3>
          </div>
          <p
            class="opacity-75 max-w-[274px] lg:max-w-[320px]"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla arcu at facilisis dignissim arcu pretium tempus. Cras scelerisque urna amet sagittis donec arcu ipsum.
          </p>
        </div>
        <div
          class="max-w-xs"
        >
          <div
            class="flex flex-row items-center gap-3 mb-5"
          >
            <div
              class="w-11 h-11 flex items-center justify-center bg-primary text-offWhite rounded-[10px]"
            >
              <svg
                fill="currentColor"
                height="1.5rem"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1.5rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M9.4 16.6L4.8 12l4.6-4.6L8 6l-6 6 6 6 1.4-1.4zm5.2 0l4.6-4.6-4.6-4.6L16 6l6 6-6 6-1.4-1.4z"
                />
              </svg>
            </div>
            <h3
              class="text-2xl font-semibold"
            >
              Service Title 2
            </h3>
          </div>
          <p
            class="opacity-75 max-w-[274px] lg:max-w-[320px]"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla arcu at facilisis dignissim arcu pretium tempus. Cras scelerisque urna amet sagittis donec arcu ipsum.
          </p>
        </div>
        <div
          class="max-w-xs"
        >
          <div
            class="flex flex-row items-center gap-3 mb-5"
          >
            <div
              class="w-11 h-11 flex items-center justify-center bg-primary text-offWhite rounded-[10px]"
            >
              <svg
                fill="currentColor"
                height="1.5rem"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1.5rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
                />
              </svg>
            </div>
            <h3
              class="text-2xl font-semibold"
            >
              Service Title 3
            </h3>
          </div>
          <p
            class="opacity-75 max-w-[274px] lg:max-w-[320px]"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla arcu at facilisis dignissim arcu pretium tempus. Cras scelerisque urna amet sagittis donec arcu ipsum.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
